Sources/TuistKit/Commands/InitCommand.swift

```diff
@@ -95,7 +95,8 @@ extension InitCommand {
     static func preprocess(_ arguments: [String]? = nil) throws {
         guard
             let arguments = arguments,
-            arguments.contains("--template")
+            arguments.contains("--template"),
+            arguments.contains("-t")
```

Comment:
What this guard statement does is check whether a custom template is specified. If not, there's no need to continue. What you've done here is make it so it only continues when the `-t` and `--template` flags are both present. This isn't what we want; the check should continue when either flag is present.

The problem with fixing this bug is that it will break the ability to use a custom template without explicitly specifying the `name` via a command-line argument.

Reply:
Shouldn't it be an or between `arguments.contains("--template")` and `arguments.contains("-t")`? 🤔

Reply:
> Shouldn't it be an or between `arguments.contains("--template")` and `arguments.contains("-t")`? 🤔

could you rephrase this? I'm not 100 % following. :)

Reply:
I mean there is a template specified if there is a --t OR a --template parameter, while right now the guard it's checking that both are present, isn't it?

Or maybe I'm just missing some context 😁

Reply:
definitely was my bad 🥲 let see if find the solution then 😄

Reply:
You should be able to just do `arguments.contains("--template") || arguments.contains("-t")`. Thanks for fixing this btw!

Reply:
That fixes the bug, but it also means commands of the form `tuist init -t templateName` won't work anymore.

Reply:
Agree with  @ezraberch , `Missing expected argument '--name <name>'` is shown when using `tuist init -t templateName` or `tuist init --template templateName`

Reply:
> but it also means commands of the form tuist init -t templateName won't work anymore.

Would you mind expanding on this one? I think I'm lacking some context here.

Reply:
The way that the `name` attribute is defined in templates is the following:

```swift
let nameAttribute: Template.Attribute = .required("name")
```

Normally, defining something as `.required` means that you must specify a value on the command line. However, there is code which overrides this requirement specifically for the `name` attribute, automatically setting it.

But there is also a second validation check (in [preprocess](https://github.com/tuist/tuist/blob/main/Sources/TuistKit/Commands/InitCommand.swift#L95)), which does not have an exception for the `name` argument. However, this check currently only runs if the `--template` argument exists. So currently `init` works and `init -t foo` works, but `init --template foo` does not.

Fixing this bug in the proposed manner will mean that the check will always run unless you're using the default template, which means you always need to specify `name` explicitly.